pkg/aggregator/aggregator.go

```diff
@@ -538,7 +538,7 @@ func (agg *BufferedAggregator) appendDefaultSeries(start time.Time, series metri
 			extra.SourceTypeName = "System"
 		}
 
-		tags := append(extra.Tags, agg.tags(false)...)
+		tags := metrics.CombineCompositeTagsAndSlice(extra.Tags, agg.tags(false))
```

Comment:
This is the only case I see of CombineCompositeTagsAndSlice in use, and it seems this is not the dogstatsd critical path (this function is only used for recurrent series, used for `datadog.agent.python.version` and `datadog.agent.check_ready`).  So, it's a bit hard to judge how this is used..

Reply:
I think `CombineCompositeTagsAndSlice` will be used only at this location. `CombineCompositeTagsAndSlice` abstracts the way to combine a `CompositeTags` and a slice which is a little tricky.